onyo/lib/filters.py

```diff
@@ -76,36 +112,235 @@ def _re_match(text: str,
         except re.error:
             return False
 
-    def match(self,
-              item: Item) -> bool:
-        r"""Does ``item`` match this ``Filter``.
+    def _tags_or_types_match(self,
+                             item: Item) -> bool | None:
+        r"""Whether the tags or types of ``self.value`` equals ``Item[self.key]``.
```

Comment:
Before I saw `_match_equal` it was unclear to me, why there's a potential `None` returned.
May be add to the docstring something like "None if not applicable"?
Alternative:
Don't return None, but a raise a dedicated error when `self.value` isn't any of these tags, so `_match_equals` can
```
try:
    return _tags_or_types_match(...)
except NewException:
    return item[self.key] == self.value
```

Reply:
Yeah, this got weird when I removed regex equality matching from `>=` and `<=`. I wanted to avoid duplicating code, but there's a lot of meaningful checks in here.

I have chosen to raise `ValueError`, as indeed the function expects a tag or an empty literal, and only reaches that point if the value is not that.

I have also added sections to the docstrings for exceptions.